esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1451,6 +1457,25 @@ EspAuthState CEspHttpServer::handleUserNameOnlyMode(EspAuthRequest& authReq)
         }
         return authFailed;
     }
+    if (!isValidUserName(userNameIn))
+    {
+        //If the username comes from the GetUserName page, the page should has the urlCookie and the askUserLogin() should be called to get the username.
+        //Otherwise, the BasicAuthenticationChallenge should be sent.
+        StringBuffer urlCookie;
+        readCookie(SESSION_START_URL_COOKIE, urlCookie);
+        if (urlCookie.isEmpty())
+        {
+            ESPLOG(LogMin, "Authentication failed (invalid username): send BasicAuthentication.");
+            m_response->sendBasicChallenge(authReq.authBinding->getChallengeRealm(), true);
```

Comment:
I'm a little unsure how we know we can send a basic challenge here. 
We have a username from:
const char* userNameIn = (authReq.requestParams) ? authReq.requestParams->queryProp("username") : NULL;
which is either a url parameter or a form parameter.  They don't have a cookie so didn't already do a GetUserName.
Is there a normal scenario where we would get here?  Or does it mean they just added username to the url?